Python/pylifecycle.c

```diff
@@ -2220,10 +2220,13 @@ add_main_module(PyInterpreterState *interp)
     }
     Py_DECREF(ann_dict);
 
-    if (_PyDict_GetItemStringWithError(d, "__builtins__") == NULL) {
-        if (PyErr_Occurred()) {
-            return _PyStatus_ERR("Failed to test __main__.__builtins__");
-        }
+    PyObject *builtins;
+    if (PyDict_GetItemStringRef(d, "__builtins__", &builtins) < 0) {
```

Comment:
It is better to use `PyDict_ContainsString()` here. Or `PyDict_Contains(_Py_ID(...))`.